src/components/ebay-progress-bar-expressive/component.ts

```diff
@@ -0,0 +1,152 @@
+import { AttrString } from "marko/tags-html";
+import type { WithNormalizedProps } from "../../global";
+
+export const messageDurationStandard = 1500;
+export const messageDurationReducedMotionMultiplier = 1.5;
+export const messageFadeInDuration = 833;
+
+interface Message {
+    text: string;
+    duration?: number;
+}
+
+interface ProgressBarExpressiveInput
+    extends Omit<Marko.Input<"div">, `on${string}`> {
+    "a11y-text"?: AttrString;
+    messages?: Marko.RepeatableAttrTag<Message>;
+    size?: "medium" | "large";
+}
+
+export interface Input
+    extends WithNormalizedProps<ProgressBarExpressiveInput> {}
+
+interface State {
+    messages: Message[];
+    messageIsFadingIn: boolean;
+    currentMessageIndex: number;
+    nextMessageIndex: number;
+    isInitialMessage: boolean;
+    useReducedMotion: boolean;
+}
+
+class ProgressBarExpressive extends Marko.Component<Input, State> {
+    declare timeouts: { fadeIn?: NodeJS.Timeout; showMessage?: NodeJS.Timeout };
+    declare fadeInFirstMessage: boolean;
+
+    onCreate(input: Input) {
+        const messages = (input.messages as Message[]) || [];
+        const useReducedMotion =
+            typeof window !== "undefined" &&
+            window.matchMedia("(prefers-reduced-motion: reduce)").matches;
+        // For medium sized text, display the first message immediately
+        this.fadeInFirstMessage = input.size !== "medium";
+        this.state = {
+            messages: messages,
+            messageIsFadingIn: false,
+            currentMessageIndex: -1,
+            nextMessageIndex: 0,
+            isInitialMessage: true,
+            useReducedMotion: useReducedMotion,
+        };
+
+        this.timeouts = { fadeIn: undefined, showMessage: undefined };
+    }
+
+    onInput(input: Input) {
+        this.state.messages = (input.messages as Message[]) || [];
+        if (this.state.nextMessageIndex >= this.state.messages.length) {
+            this.state.nextMessageIndex = 0;
+        }
+
+        this.initializeMessageRotation();
```

Comment:
Lets move this to 
`onUpdate` and `onMount`

We should have cleanups on `onRender` in addition to `onDestroy`

See https://github.com/eBay/ebayui-core/blob/master/src/components/ebay-listbox-button/component.ts#L79-L95
(It might not be needed, but I think its good practice in case the parent rerenders or such)

Reply:
Can you explain this a little more? We only need to re-initialize the message rotation when the messages input is set/changed, so it seems like `onInput` would be the right place for this. `onUpdate` and `onRender` get called frequently for this component, since CSS classes and text content change based on state.